### ISSUE TITLE
Log HTTPs cert info on startup

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/AspNetCore#4759]
Kestrel logs endpoint when the server starts, but doesn't log which certificates were used. Now that Kestrel endpoints and certs can be configured from a file https://github.com/aspnet/KestrelHttpServer/pull/2186, it would be nice to show console info to help users identify if their config was correctly used.

Some ideas:

From store, show friendly name

    Now listening on: https://localhost:5001
    Using server certificate: "ASP.NET Core HTTPS development certificate" from cert:\CurrentUser\My

From store, show subject

    Now listening on: https://localhost:5001
    Using server certificate: CN=localhost, cert:\CurrentUser\My

From store, include more info like thumbprint

    Now listening on: https://localhost:5001
    Using server certificate: CN=localhost, cert:\CurrentUser\My, 52A477BBEDE8DFDEB699106D5FFB8FE89F9BF790

From file

    Now listening on: https://localhost:5001
    Using server certificate: C:\dev\test\cert.pfx

Multiple certs

    Now listening on: https://localhost:5002
    Using server certificate: C:\dev\test\cert2.pfx
    Now listening on: https://localhost:5001
    Using server certificate: C:\dev\test\cert1.pfx

**Comment by somy55** [comment]
F

**Comment by davidfowl** [comment]
This seems like a good thing,  thoughts @blowdart ?

**Comment by blowdart** [comment]
Aside from thumbprint yes (thumbprint is sha1, we don't like sha1)

Also add valid from and valid to or maybe the san (or first 5 hosts in the ssn, because those can get big)

**Comment by BrennanConroy** [comment]
Triage: This would be nice.
Issues: Certs wouldn't necessarily line up with the address they apply to. Can we get the store they came from?

`Log.LogInfo("Loaded cert from X; <cert info>")`

**Comment by davidfowl** [comment]
> Issues: Certs wouldn't necessarily line up with the address they apply to. Can we get the store they came from?

This is because hosting prints the addresses right?

**Comment by BrennanConroy** [comment]
> This is because hosting prints the addresses right?

Right. https://github.com/dotnet/aspnetcore/issues/23601 is related, if we have some way of grouping startup logs to make the output nice.

**Comment by adityamandaleeka** [comment]
@ldillonel This looks like it's relevant to your cert work. Might be good to look into it.

**Comment by Pilchard123** [comment]
> thumbprint is sha1, we don't like sha1

Why is using SHA1 for thumbprints a problem here @blowdart ? AIUI, thumbprints aren't used for any security purpose, merely for making it easier to refer to a particular certificate. If it was used for signatures or the like, then sure, but I don't think there's any security issue in using it as an identifier.

**Comment by blowdart** [comment]
It's a problem because we're trying to eliminate all use of sha1 internally, no matter what it's used for. We have a lot of automated tooling that helps us avoid mistakes and the tooling doesn't understand context. We also have customers who run their own tools and flag of any use of sha1 and then I end up with emails and having to explain the context that no, this isn't a cryptographic use. And if it's used over certificates it makes that explanation so much harder.

So it's simpler to eliminate all uses. It's the main reason we have non-cryptographic hash algorithms now.
